js_modules/dagit/packages/core/src/assets/AssetPartitions.tsx

```diff
@@ -200,18 +233,39 @@ export const AssetPartitions: React.FC<Props> = ({
             flex={{direction: 'column'}}
             border={{side: 'right', color: Colors.KeylineGray, width: 1}}
             background={Colors.Gray50}
+            data-testid={testId(`partitions-${selection.dimension.name}`)}
           >
-            {selection.dimension.name !== 'default' && (
-              <Box
-                padding={{horizontal: 24, vertical: 8}}
-                flex={{gap: 8, alignItems: 'center'}}
-                background={Colors.White}
-                border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+            <Box
+              flex={{direction: 'row', justifyContent: 'space-between', alignItems: 'center'}}
+              background={Colors.White}
+              border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+              padding={{horizontal: 24, vertical: 8}}
+            >
+              <div>
+                {selection.dimension.name !== 'default' && (
+                  <Box flex={{gap: 8, alignItems: 'center'}}>
+                    <Icon name="partition" />
+                    <Subheading>{selection.dimension.name}</Subheading>
+                  </Box>
+                )}
+              </div>
+              <SortButton
```

Comment:
It might break out design standard a bit but could we push the button further to the right so it's lodged in the corner nicely above the status dots? It seems a little too padded at the moment